### ISSUE TITLE
Checks are much slower with (some?) SlevomatCodingStandard rules

### ISSUE THREAD

**Issue by ocean90** [wearerequired/coding-standards#59]
Example file: [inc/Blocks/namespace.php](https://github.com/wearerequired/swisscom-content-types/blob/8962299403e4a3c5403bb2b351cbd99156f74211/inc/Blocks/namespace.php)

With [current](https://github.com/wearerequired/coding-standards/blob/a70608d944cc530c886025bbcdb0683e5f084463/Required/ruleset.xml#L1) rules:

	$ phpcs inc/Blocks/namespace.php
	Time: 11.11 secs; Memory: 12MB

Without any SlevomatCodingStandard rules:

	$ phpcs inc/Blocks/namespace.php
	Time: 511ms; Memory: 12MB

This is also noticeable in VS Code where checks are delayed and two PHP processes constantly running with high CPU usage.

**Comment by derpaschi** [comment]
I can copy that.

Example in 💩 code with [current](https://github.com/wearerequired/coding-standards/blob/a70608d944cc530c886025bbcdb0683e5f084463/Required/ruleset.xml#L103-L169) rules.

```
phpcs includes/models/JobModel.php
Time: 13.38 secs; Memory: 14MB
```

Without any SlevomatCodingStandard rules (same file):
```
phpcs includes/models/JobModel.php
Time: 770ms; Memory: 14MB
```

**Comment by derpaschi** [comment]
I've identified 2 rules which are causing this.

```
<!-- All references to types named Exception or ending with Exception must be referenced via a fully qualified name. -->
<!-- 5 - 6 seconds -->
<rule ref="SlevomatCodingStandard.Namespaces.FullyQualifiedExceptions"/>

<!-- Looks for unused imports from other namespaces. -->
<!-- 6 - 7 seconds -->
<rule ref="SlevomatCodingStandard.Namespaces.UnusedUses"/>
```

Same code, without this 2 rules:
```
> phpcs includes/models/JobModel.php --no-cache
Time: 716ms; Memory: 12MB
```

**Comment by ocean90** [comment]
Danke, war bei mir auch zu und habe dann kurzen Prozess gemacht und die beiden Regeln erstmal deaktiviert.

**Comment by ocean90** [comment]
Upstream issue: https://github.com/slevomat/coding-standard/issues/929

**Comment by kukulich** [comment]
The file `inc/Blocks/namespace.php` is private ¯\_(ツ)_/¯

**Comment by ocean90** [comment]
@kukulich Happy to share the file with you. What's you email address? Or feel free to drop me a line dominik(at)required.com.

**Comment by ocean90** [comment]
The slow sniffs have been fixed so we can re-add them once 6.2.0 is available.
